FreeRTOS/Demo/RISC-V-Qemu-virt_GCC/main.c

```diff
@@ -0,0 +1,231 @@
+#define mainQUEUE_LENGTH					( 1 )
+
+/*
+ * Prototypes for the standard FreeRTOS callback/hook functions implemented
+ * within this file.  See https://www.freertos.org/a00016.html
+ */
+void vApplicationMallocFailedHook( void );
+void vApplicationIdleHook( void );
+void vApplicationStackOverflowHook( TaskHandle_t pxTask, char *pcTaskName );
+void vApplicationTickHook( void );
+
+/*-----------------------------------------------------------*/
+
+/* The queue used by both tasks. */
+static QueueHandle_t xQueue = NULL;
+
+/*-----------------------------------------------------------*/
+
+static void prvQueueSendTask( void *pvParameters )
+{
+TickType_t xNextWakeTime;
+const unsigned long ulValueToSend = 100UL;
+const char * const pcMessage1 = "Transfer1";
+const char * const pcMessage2 = "Transfer2";
+int f = 1;
+
+	/* Remove compiler warning about unused parameter. */
+	( void ) pvParameters;
+
+	/* Initialise xNextWakeTime - this only needs to be done once. */
+	xNextWakeTime = xTaskGetTickCount();
+
+	for( ;; )
+	{
+		char buf[40];
+
+		sprintf( buf, "%d: %s: %s", xGetCoreID(),
+				pcTaskGetName( xTaskGetCurrentTaskHandle() ),
+				( f ) ? pcMessage1 : pcMessage2 );
+		vSendString( buf );
+		f = !f;
+
+		/* Place this task in the blocked state until it is time to run again. */
+		vTaskDelayUntil( &xNextWakeTime, mainQUEUE_SEND_FREQUENCY_MS );
+
+		/* Send to the queue - causing the queue receive task to unblock and
+		toggle the LED.  0 is used as the block time so the sending operation
+		will not block - it shouldn't need to block as the queue should always
+		be empty at this point in the code. */
+		xQueueSend( xQueue, &ulValueToSend, 0U );
+	}
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvQueueReceiveTask( void *pvParameters )
+{
+unsigned long ulReceivedValue;
+const unsigned long ulExpectedValue = 100UL;
+const char * const pcMessage1 = "Blink1";
+const char * const pcMessage2 = "Blink2";
+const char * const pcFailMessage = "Unexpected value received\r\n";
+int f = 1;
+
+	/* Remove compiler warning about unused parameter. */
+	( void ) pvParameters;
+
+	for( ;; )
+	{
+		char buf[40];
+
+		/* Wait until something arrives in the queue - this task will block
+		indefinitely provided INCLUDE_vTaskSuspend is set to 1 in
+		FreeRTOSConfig.h. */
+		xQueueReceive( xQueue, &ulReceivedValue, portMAX_DELAY );
+
+		/*  To get here something must have been received from the queue, but
+		is it the expected value?  If it is, toggle the LED. */
+		if( ulReceivedValue == ulExpectedValue )
+		{
+			sprintf( buf, "%d: %s: %s", xGetCoreID(),
+					pcTaskGetName( xTaskGetCurrentTaskHandle() ),
+					( f ) ? pcMessage1 : pcMessage2 );
+			vSendString( buf );
+			f = !f;
+
+			ulReceivedValue = 0U;
+		}
+		else
+		{
+			vSendString( pcFailMessage );
+		}
+	}
+}
+
+/*-----------------------------------------------------------*/
+
+int main( void )
+{
+	vSendString( "Hello FreeRTOS!" );
+
+	/* Create the queue. */
```

Comment:
could you please separate the main file from the file that actually do the demo
this way if someone wants to add more demos, their job will be easier, by just calling the appropriate function 
```c
#if defined BLINKY
    main_blinky()
#elif defined NETWORKING
    main_networking()
#endif 
```
for example